src/components/Link/Link.test.tsx

```diff
@@ -1,19 +1,143 @@
+  const weight = faker.helpers.arrayElement(
+    Object.keys(fontWeights)
+  ) as keyof typeof fontWeights;
 
-test.todo("adds rel='noopener noreferrer' when target is '_blank'");
+  render(<Basic weight={weight} />);
+  expect(screen.getByRole("link")).toHaveClass(`fw_${weight}`);
+});
 
-test.todo("can be used with Icon component");
+test("applies the correct styles for low and high contrast", () => {
+  render(<Contrasts />);
+
+  expect(screen.getByRole("link", { name: "낮은 명암비" })).toHaveClass(
+    "c_text.muted"
+  );
+
+  expect(screen.getByRole("link", { name: "높은 명암비" })).toHaveClass(
+    "c_text"
+  );
+});
+
+test("applies the correct underline styles", () => {
+  render(<Underlines />);
+
+  expect(screen.getByRole("link", { name: "밑줄 있음" })).toHaveClass(
+    "td_underline"
+  );
+  expect(screen.getByRole("link", { name: "밑줄 없음" })).toHaveClass(
+    "td_none"
+  );
+});
+
+test("forwards additional anchor props like 'href' and 'target'", () => {
+  const href = faker.internet.url({ appendSlash: true });
+  const target = "_self";
+
+  render(<Basic href={href} target={target} />);
+
+  const link = screen.getByRole("link");
+  expect(link).toHaveAttribute("href", href);
+  expect(link).toHaveAttribute("target", target);
+});
+
+test("adds rel='noopener noreferrer' when target is '_blank'", () => {
+  render(<Security />);
+
+  expect(
+    screen.getByRole("link", { name: "새 탭에서 열기 (보안 속성 자동 추가)" })
+  ).toHaveAttribute("rel", "noopener noreferrer");
+
+  expect(
+    screen.getByRole("link", { name: "같은 탭에서 열기" })
+  ).not.toHaveAttribute("rel", "noopener noreferrer");
+});
+
+test("can be used with Icon component", () => {
+  render(<WithIcon />);
+
+  const link = screen.getByRole("link");
+  expect(link).toHaveTextContent("링크");
+  expect(link.querySelector("svg")).toBeInTheDocument();
+});
+
+test("navigates to the correct URL when clicked", async () => {
+  const href = faker.internet.url({ appendSlash: true });
+  render(<Basic href={href} onClick={undefined} />);
+
+  const link = screen.getByRole("link");
+  expect(link).toHaveAttribute("href", href);
+  await userEvent.click(link);
+
+  expect(window.location.href).toBe(href);
+});
+
+test("handles click events correctly", async () => {
+  const handleClick = vi.fn();
+  render(<Basic href="#" onClick={handleClick} />);
+
+  const link = screen.getByRole("link");
+  await userEvent.click(link);
+
+  expect(handleClick).toHaveBeenCalledTimes(1);
+});
+
+test("link is keyboard focusable", () => {
+  const href = faker.internet.url({ appendSlash: true });
+  render(<Basic href={href} />);
+
+  const link = screen.getByRole("link");
+  link.focus();
+  expect(link).toHaveFocus();
+});
+
+test("navigates when link is clicked with keyboard", async () => {
+  const href = faker.internet.url({ appendSlash: true });
+  render(<Basic href={href} onClick={undefined} />);
+
+  const link = screen.getByRole("link");
+  link.focus();
```

Comment:
키보드 사용자는 이런 식으로 링크로 선택할 수 없습니다.

```suggestion
  await userEvent.tab()
```

@hyoseong1994 님, 아직도 뭔가 시간이 쫓기어 급하게 작업하시는 느낌이 들어요... 😭
아무도 @hyoseong1994 님께 빨리 개발하시라고 독촉하지 않습니다. 필요할 때는 천천히 학습을 하시는 기쁨도 좀 느끼시면서 프로젝트를 하셨으면 좋겠습니다. 여유를 좀 가지시고 제가 공유한 블로그 글이나 유튜브 영상을 한 번 보시기를 추천드립니다. 🙏

Reply:
넵 죄송합니다. 천천히 학습하는 마음으로 작업하도록하겠습니다.
공유한 내용 학습 후 작업 진행하도록하겠습니다.